### ISSUE TITLE
default namespace roots

### ISSUE THREAD

**Issue by cosnicolaou** [vanadium/core#45]
Ideally it should be possible to specify default namespace roots from a per installation data file that's easy to version control locally.

**Comment by siddharthab** [comment]
This becomes even more important in the context of Go Modules. One idea could be to have a separate vanadium defaults module that people can `replace` in their go.mod file without having to replace entire v.io.

**Comment by cosnicolaou** [comment]
unfortunately I don't see a way to achieve the goal of having the tools in
v.io (e.g. namespaced) use grail specific defaults without changing the v.io
package itself. GRAIL could chose to rely on environment variables to set
the defaults instead of having them be hardcoded. All other schemes are
essentially equivalent to environment variables.

Can you think of a cleaner/better solution?

Cheers, Cos.

**Comment by siddharthab** [comment]
In the world of Go modules, I think the best way forward is to have a separate module for default settings. More concretely,

We start a new github repository, say github.com/vanadium/site-defaults with an import path as v.io/x/ref/lib/flags/sitedefaults, with contents as a single file sitedefaults.go
```go
// Package sitedefaults contains the default values for core Vanadium flags.
// These can be overridden at an install site by either editing the vendored
// copy or using the replace directive in the go.mod file of the main module.
package sitedefaults

var SiteDefaults = []map[string]interface{}{{
	"namespaceRoots": []string{
		"/(dev.v.io:r:vprod:service:mounttabled)@ns.dev.v.io:8101",
	},
}}
```

And then in v.io/x/ref/lib/flags/zzz.go, we import v.io/x/ref/lib/flags/sitedefaults, and merge in the values from SiteDefaults.

What do you think?

**Comment by cosnicolaou** [comment]
ah ok, good idea, this works for me, let me get a revision out to you
hopefully today.

Cheers, Cos.

**Comment by siddharthab** [comment]
This issue can be considered resolved after #84 and #87.
